### PR TITLE
Function test. Verify if extra ahci controller added to mount drives more than 6

### PR DESCRIPTION
**Pull request by JuneZhou19**
For HWSIM-1216. @fub2 @xiar @XiaowenJiang @chenge3 @turtle-fly

### CODE REVIEW

**Comment by turtle-fly** [comment]
@JuneZhou19 there is a behavior definition in story: "product code: warning if max number drives beyond 6". We may understand the feature per controller, and implement this feature accordingly.

**Comment by emcinfrasim** [comment]
```
******* Functional Test ******* 
TestName: test_controller_with_more_than_drive6 
ErrorType: exceptions.AssertionError 
ErrorMessage: 1 != 2
-------------------- >> begin captured stdout << ---------------------
[ 15789  ] test-socat starts to run
[ 15793  ] test-bmc starts to run
[ 15802  ] test-node is running
[ 15813  ] test-racadm starts to run

--------------------- >> end captured stdout << ----------------------
-------------------- >> begin captured logging << --------------------
root: INFO: Creating drive: 
root: INFO: Creating drive: 
root: INFO: Creating drive: 
root: INFO: Creating drive: 
root: INFO: Creating drive: 
root: INFO: Creating drive: 
root: INFO: Creating drive: 
root: INFO: Creating drive: 
root: INFO: Creating drive: 
root: INFO: Creating drive: 
root: INFO: Creating drive: 
root: INFO: /usr/bin/socat pty,link=/home/infrasim/.infrasim/test/.pty0,waitslave unix-listen:/home/infrasim/.infrasim/test/.serial,fork
root: INFO: ipmi_sim -c /home/infrasim/.infrasim/test/etc/vbmc.conf -f /home/infrasim/.infrasim/test/data/dell_r730.emu -n -s /home/infrasim/.infrasim/test/data
root: INFO: racadmsim test 0.0.0.0 10022 admin admin auto
--------------------- >> end captured logging << --------------------- 
```

**Comment by JuneZhou19** [comment]
It captured a bug in "infrasim node info" command that the extra ahci not shown, so it's expected to fail.

**Comment by turtle-fly** [comment]
@JuneZhou19 good catch! Please help fix the node info errors

**Comment by JuneZhou19** [comment]
No problem. Will the failure block PR from merging? @turtle-fly

**Comment by turtle-fly** [comment]
@JuneZhou19 all PR goes with all functional and unit tests pass~

**Comment by emcinfrasim** [comment]
```
******* Functional Test ******* 
TestName: test_controller_with_more_than_drive6 
ErrorType: exceptions.AssertionError 
ErrorMessage: 1 != 2
-------------------- >> begin captured stdout << ---------------------
[ 14659  ] test-socat starts to run
[ 14662  ] test-bmc starts to run
[ 14671  ] test-node is running
[ 14682  ] test-racadm starts to run

--------------------- >> end captured stdout << ----------------------
-------------------- >> begin captured logging << --------------------
root: INFO: Creating drive: 
root: INFO: Creating drive: 
root: INFO: Creating drive: 
root: INFO: Creating drive: 
root: INFO: Creating drive: 
root: INFO: Creating drive: 
root: INFO: Creating drive: 
root: INFO: Creating drive: 
root: INFO: Creating drive: 
root: INFO: Creating drive: 
root: INFO: /usr/bin/socat pty,link=/home/infrasim/.infrasim/test/.pty0,waitslave unix-listen:/home/infrasim/.infrasim/test/.serial,fork
root: INFO: ipmi_sim -c /home/infrasim/.infrasim/test/etc/vbmc.conf -f /home/infrasim/.infrasim/test/data/dell_r730.emu -n -s /home/infrasim/.infrasim/test/data
root: INFO: racadmsim test 0.0.0.0 10022 admin admin auto
--------------------- >> end captured logging << --------------------- 
```

**Comment by JuneZhou19** [comment]
retest this please

**Comment by JuneZhou19** [comment]
@turtle-fly about behavior definition on product code warning, since controller only support no more than 6 drives which is the HW behavior, and we are simulating the behavior. So it should be irrelevant with our code.
"Infrasim node info" issue fixed.
Added some more verification in last 2 PRs.

**Comment by JuneZhou19** [comment]
All commits submitted for this PR.

File: test/functional/test_storage.py [hunk 1]
```diff
@@ -194,6 +194,153 @@ def test_controller_with_drive2(self):
...
+        assert "/tmp/sdc.img" in qemu_cmdline
+        assert "/tmp/sdd.img" in qemu_cmdline
+        assert "/tmp/sde.img" in qemu_cmdline
+        assert "/tmp/sdf.img" in qemu_cmdline
+        assert "/tmp/sdg.img" in qemu_cmdline
+        assert "format=qcow2" in qemu_cmdline
+        assert "-device ahci,id=sata1" in qemu_cmdline
+        assert "-device ahci,id=sata0" in qemu_cmdline
+        assert "drive=sata0-0-5-0" in qemu_cmdline
+        assert "drive=sata1-0-0-0" in qemu_cmdline
```
Comment (turtle-fly) [inline test/functional/test_storage.py]: :+1:

**Comment by chenge3** [comment]
:+1:

**Pull request merged**